frontend/src/components/ConstraintForm/ConstraintForm.vue

```diff
@@ -0,0 +1,70 @@
+<script>
+import { AccordionList } from 'vue3-rich-accordion';
+import ClassAccordionItem from './ClassAccordionItem.vue';
+import { useFormStore } from '@/stores/form';
+
+export default {
+  setup() {
+    const formStore = useFormStore();
+
+    const generateTimetable = () => {
+      // TODO: submit handler will be implemented when algorithm backend is merged
+      console.log(formStore.form);
+    };
+
+    return { formStore, generateTimetable };
+  },
+  components: {
+    AccordionList,
+    ClassAccordionItem,
+  },
+};
+</script>
+
+<template>
+  <form @submit.prevent="generateTimetable">
+    <h2>Classes</h2>
+    <accordion-list
+      @update:state="newState => (formStore.accordionState = newState)"
+      open-multiple-items
+      :state="formStore.accordionState"
+    >
+      <class-accordion-item v-for="(_, classIndex) in formStore.form.classes" :key="classIndex" :index="classIndex" />
```

Comment:
I believe [this](https://vuejs.org/guide/essentials/list.html#v-for-with-a-range) approach would be more valid for this use case.